library/infoblox_host_record.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+# Copyright (c) 2017 Philip Bove
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = """
+module: infoblox_host_record
+description: manage infoblox host records with infoblox-client
+version_added: "2.4"
+"""
+
+EXAMPLES = """
+- name: create host_record
+  host: "{{grid_manager}}"
+  name: server.test.com
+  mac: macaddr
+  ip_address: 10.1.10.50
+  username: test
+  password: test
+"""
+from ansible.module_utils.basic import *
+
+try:
+	from infoblox_client import objects, connector, exceptions
+except ImportError:
+	raise Exception('infoblox-client is not installed.  Please see details here: https://github.com/infobloxopen/infoblox-client')
+
+def ensure(module):
+	try:
+		conn = connector.Connector({'host':module.params['host'],'username':module.params['username'],'password':module.params['password'],
+			'ssl_verify':module.params['validate_certs'],'wapi_version':module.params['wapi_version']})
+		ip_addr = ip_gen(module)
+		host_record = objects.HostRecord.search(conn, name=module.params['name'], view=module.params['dns_view'])
+		if module.params['state'] == 'present':
+			create_host_record(conn, host_record, module, ip_addr)
+		elif module.params['state'] == 'absent':
+			delete_host_record(conn, host_record, module)
+	except exceptions.InfobloxException as error:
+		module.fail_json(msg=str(error))
+
+def delete_host_record(conn, host_record, module):
+	try:
+		if host_record:
+			host_record.delete()
+			module.exit_json(changed=True)
+		module.exit_json(changed=False)
+	except exceptions.InfobloxException as error:
+		module.fail_json(msg=str(error))
+
+def create_host_record(conn,host_record, module, ip_addr):
+	try:
+		#If host record exists check if there is a difference between what is specified in ansible
+		#update_if_exists=True seems to no actually do anything and just errors with (record already exists)
+		#using a delete/create method to "update"
```

Comment:
Seems unsafe, if I remember correctly the library has ability to update as approriate. We should be able to make a generic helper class within the module to do something like check what is present and diff, and only push that.

Reply:
I agree, I was trying to use the infoblox-client's "update_if_exists" in an earlier iteration but it kept throwing "record already exists". I'll give that another try, I was probably doing something wrong.

Reply:
I can confirm that "update_if_exists=True" does not seem to work as intended for Host Records. Maybe this is because I'm on an older Infoblox version (WAPI version 1.7.1)?

I've confirmed it is hitting it for an update:

https://github.com/bandit145/infoblox-ansible/blob/5cd8380afcc8122bc9efe2991d21fc3d7aa8e2d0/library/infoblox_host_record.py#L74

Reply:
Can you open an issue there?

Reply:
I was going to open an issue but found this https://github.com/infobloxopen/infoblox-client/issues/68 (Not sure how I didn't see it before). I tested just a comment change and that worked fine.

I'm not really sure how to handle that in the module, just ignore any IP changes?